src/components/create/SendBitcoin.vue

```diff
@@ -206,11 +210,25 @@ export default class SendBitcoin extends Vue {
       });
   }
 
-  @Emit('back')
-  async back(currentComponent: 'ConnectDevice' | 'PegInForm') {
+  @Emit()
+  async backToConnectDevice() {
     await this.clear();
     this.clearAccount();
-    return currentComponent;
+    let wallet: BtcWallet;
+    if (this.peginTxState.bitcoinWallet) {
+      wallet = this.peginTxState.bitcoinWallet;
+    } else {
+      await this.back();
+    }
+    this.clearStore();
+    this.init()
+      .then(() => this.setBtcWallet(wallet));
+  }
+
+  @Emit('back')
+  back() {
+    this.stopAskingForBalance()
+      .then(() => this.$router.push({ name: 'Home' }));
```

Comment:
Seems like back event is not required anymore... as this looks like _go home_ event